### PR TITLE
Upgrade Meilisearch project to .NET Standard 2.1

### LINKED ISSUE #221: Drop .NET 2.0 support, upgrade the minimum to .NET 2.1
Following this discussion: https://github.com/meilisearch/meilisearch-dotnet/issues/218

After some problems during the implementation of the v0.25, we decided internally that we should drop support of .net 2. The main reason is not based only on the v0.25 implementation problems but basically because we are supporting a version that is not supported even by Microsoft anymore.

The proposal here is to follow the .net core guidelines about EOL:
https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core

Upgrade `meilisearch-dotnet` to ensure minimum support of `.NET Core 3.1`

- [ ] Update CI
- [ ] Update source code/tests
- [ ] Update README, CONTRIBUTING files

These concerns were raised originally here: #202 https://github.com/meilisearch/meilisearch-dotnet/issues/218#issuecomment-1020101386

### PR DESCRIPTION
**Pull request by juchom**
# Pull Request

## What does this PR do?
Fixes #221 

Upgrade to .NET Standard 2.1

## PR checklist
Please check if your PR fulfills the following requirements:
- [X] Does this PR fix an existing issue?
- [X] Have you read the contributing guidelines?
- [X] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to MeiliSearch!

### CODE REVIEW

File: src/Meilisearch/Meilisearch.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFramework>netstandard2.0</TargetFramework>
+        <TargetFramework>netstandard2.1</TargetFramework>
```
Comment (brunoocasali) [inline src/Meilisearch/Meilisearch.csproj:4]: @juchom do you think is good to tell people which .net version this package supports in the README - being explicitly about that?

**Review by brunoocasali — COMMENTED** [review]
Just a comment regarding adding references about supported version in the README.

File: src/Meilisearch/Meilisearch.csproj (hunk 1 above)
Comment (juchom) [reply src/Meilisearch/Meilisearch.csproj:4]: Good question, I went to a couple of repositories and this is not specified. I don't have any opinion on this.

If you want to this to the README, just give me the text and I'll update the issue.

Comment (brunoocasali) [reply src/Meilisearch/Meilisearch.csproj:4]: Awesome @juchom 💯 

You could add in line 48 of `README.md` this text:

This package targets .NET Standard 2.1.

**Review by brunoocasali — APPROVED** [review]
Thanks again @juchom!! ❤️ 

LGTM!

**Comment by brunoocasali** [comment]
bors merge

**Pull request merged**